providers/edge3/src/airflow/providers/edge3/worker_api/routes/ui.py

```diff
@@ -100,3 +105,57 @@ def jobs(
         jobs=result,
         total_entries=len(result),
     )
+
+
+class MaintenanceRequest(BaseModel):
+    """Request body for maintenance operations."""
+
+    maintenance_comment: Annotated[str, Field(description="Comment describing the maintenance reason.")]
+
+
+@ui_router.post(
+    "/worker/{worker_name}/maintenance",
+)
```

Comment:
To ensure this API is not "public/open" but authenticated, please add the access check
```suggestion
    dependencies=[
        Depends(requires_access_view(access_view=AccessView.JOBS)),
    ],
)
```

Reply:
@jscheffl , I already tried this and I was not able to get the post and delete to work. I keep getting 401 not authorized and I was not able to figure out how to provide proper authorization.

Reply:
Good. But still needed. Probably because you implemented the manual authentication in the service accessor.
Can you attempt it via the axios generated code? If then not working, please push and I can help fixing.

Reply:
Just added this in commit https://github.com/apache/airflow/pull/55348/commits/ec77bd65b96fd27eb601aeda51d4fb92354bc301 and it worked for me.